### PR TITLE
Add management command to delete inactive users

### PR DESCRIPTION
**Pull request by mhieta**
Delete users inactive for INACTIVE_USER_DELETION_YEARS years (default 5), falling back to date_joined when last_api_use is null. Users with active signups (attending, awaiting payment, or waitlisted) are skipped.

Includes tests for all scenarios.

Refs: [LINK-2196](https://helsinkisolutionoffice.atlassian.net/browse/LINK-2196)

[LINK-2196]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-2196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Automated removal of inactive user accounts using the most recent activity timestamp (API use, last login, or account creation).
  * Configurable inactivity threshold (default 5 years) with validation to reject invalid values.
  * Skips and preserves users with active signup/event relationships and reports deleted vs skipped counts.

* **Tests**
  * Comprehensive tests for deletion behavior, setting validation, timestamp fallbacks, signup protections, boundary cases, and privileged-account handling.

### CODE REVIEW

File: events/management/commands/delete_inactive_users.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        threshold_time = now - relativedelta(
+            years=settings.INACTIVE_USER_DELETION_YEARS
+        )
+
+        # For users without a last_api_use timestamp, fall back to date_joined.
+        candidates = (
+            User.objects.annotate(
+                last_activity=Coalesce(
+                    "last_api_use", "date_joined", output_field=DateTimeField()
```
Comment (charn) [inline events/management/commands/delete_inactive_users.py]: `last_api_use` was added in a commit made in 10.9.2025. If the default 5 years is used then there's a risk that users are deleted way before this 5 year threshold.

File: events/management/commands/delete_inactive_users.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    def handle(self, *args, **options):
+        now = timezone.now()
+
+        threshold_time = now - relativedelta(
+            years=settings.INACTIVE_USER_DELETION_YEARS
+        )
+
+        # For users without a last_api_use timestamp, fall back to date_joined.
+        candidates = (
```
Comment (charn) [inline events/management/commands/delete_inactive_users.py]: Just in case I'll add a note that Django admin users use `last_login` field and without considering that field those users might get deleted 5 years after being created.

Comment (mhieta) [reply events/management/commands/delete_inactive_users.py]: Added `last_login`-field to candidate evaluation process.

File: events/management/commands/delete_inactive_users.py (hunk 1 above)
Comment (mhieta) [reply events/management/commands/delete_inactive_users.py]: The `date_joined` field is auto-generated on User-instance creation phase so that should prevent under 5 years old user deletion by default.

File: events/tests/management/test_delete_inactive_users.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    now = localtime()
+    expiration_years = django_settings.INACTIVE_USER_DELETION_YEARS
+    expired_time = now - relativedelta(years=expiration_years + 1)
+
+    creator = UserFactory(last_api_use=expired_time)
+    other_user = UserFactory(last_api_use=expired_time)
+
+    SignUpFactory(
+        created_by=creator,
+        attendee_status=SignUp.AttendeeStatus.WAITING_LIST,
```
Comment (charn) [inline events/tests/management/test_delete_inactive_users.py]: Could use `pytest.mark.parametrize` here to go through all the active statuses.

File: events/tests/management/test_delete_inactive_users.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    expiration_years = django_settings.INACTIVE_USER_DELETION_YEARS
+    expired_time = now - relativedelta(years=expiration_years + 1)
+
+    user_with_group = UserFactory(last_api_use=expired_time)
+    user_without_group = UserFactory(last_api_use=expired_time)
+
+    group = SignUpGroupFactory()
+    SignUpFactory(
+        signup_group=group,
+        attendee_status=SignUp.AttendeeStatus.ATTENDING,
```
Comment (charn) [inline events/tests/management/test_delete_inactive_users.py]: Could use `pytest.mark.parametrize` here to go through all the active statuses.

File: events/tests/management/test_delete_inactive_users.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+@pytest.mark.django_db
+def test_delete_inactive_users_skips_user_with_active_signup_as_contact_person():
+    now = localtime()
+    expiration_years = django_settings.INACTIVE_USER_DELETION_YEARS
+    expired_time = now - relativedelta(years=expiration_years + 1)
+
+    user_with_signup = UserFactory(last_api_use=expired_time)
+    user_without_signup = UserFactory(last_api_use=expired_time)
+
+    signup = SignUpFactory(attendee_status=SignUp.AttendeeStatus.ATTENDING)
```
Comment (charn) [inline events/tests/management/test_delete_inactive_users.py]: Could use `pytest.mark.parametrize` here to go through all the active statuses.

**Comment by voneiden** [comment]
@coderabbitai review

File: events/management/commands/delete_inactive_users.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            last_activity=Coalesce(
+                Greatest(
+                    Coalesce(last_api_use_dt, "last_login"),
+                    Coalesce("last_login", last_api_use_dt),
+                    output_field=DateTimeField(),
+                ),
+                "date_joined",
+                output_field=DateTimeField(),
+            )
+        ).filter(last_activity__lt=threshold_time)
```
[the remark this answers is not in the document]
Comment (mhieta) [reply events/management/commands/delete_inactive_users.py]: Deleting inactive superuser and staff accounts is intentional, because they also need to be deleted after inactivity.

[the remark this answers is not in the document]
Comment (mhieta) [reply events/management/commands/delete_inactive_users.py]: Will add spesific tests for these privileged users to make the usecases more clear.

**Review by charn — APPROVED** [review]
LGTM! :+1: :shipit:
